### ISSUE TITLE
Setting AutoEllipsis property as true, tooltip still appears after running project when no ellipsis is shown in design time

### ISSUE THREAD

**Issue by Marie-Shi** [dotnet/winforms#5527]
* .NET Core Version: 
.Net 6.0.100-rc.1.21417.19

* Have you experienced this same bug with .NET Framework?: 
Yes

**More info:**
1.	This issue can also repro on .Net Framework project.
2.	This issue can also repro on .Net 3.1 and .Net 5.0.

**Minimal repro:**
1.	Extract the “WinFormsApp3.zip” and open it in VS.
[WinFormsApp3.zip](https://github.com/dotnet/winforms/files/7044458/WinFormsApp3.zip)
2.	Build and run this application.
3.	Move the mouse hover over button1 and button2, then observe the form.

**Actual behavior:**
Tooltip still appears when moving the mouse hover over button1.
![Untitled](https://user-images.githubusercontent.com/45864017/130742803-e8982b09-e795-4ffd-81e0-39e9e3abafff.png)
![customer_issueupdate](https://user-images.githubusercontent.com/45864017/130742812-49b7a1b5-3f0b-4b1e-8ab4-5dc440d1f154.gif)

**Expected behavior:**
Tooltip doesn’t appear when moving the mouse hover over button.

**Comment by elachlan** [comment]
This is where the tooltip is shown.
https://github.com/dotnet/winforms/blob/3707cb85cf8f97107ca1d84a0a0fb464ea9c5f22/src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs#L938-L953

I believe you would have to use `TextRenderer.MeasureText` and compare it with the button size minus the margins. The condition would have to be updated to check that when `AutoEllipsis` is set.

I think #7341 needs to be fixed for this method to be effective, since that issue is an issue with `TextRenderer.MeasureText`.

**Comment by merriemcgaw** [comment]
> I think https://github.com/dotnet/winforms/issues/7341 needs to be fixed for this method to be effective, since that issue is an issue with TextRenderer.MeasureText.

I agree 💯 with you on that. I'll bring it up in today's meeting, but this particular issue isn't going to make it super high on the team's list. Obviously, community assistance is always appreciated.

**Comment by elachlan** [comment]
Once the issue with `TextRenderer.MeasureText` is resolved I am happy to submit a fix for this.

**Comment by merriemcgaw** [comment]
I will work with @JeremyKuhne if he or @lonitra can dig through the old code and figure out the MeasureText issue.

**Comment by elachlan** [comment]
Based on https://github.com/dotnet/winforms/issues/9672#issuecomment-1668447451 I think that measure text is working fine, we just need to specify the font. So we can proceed with a fix.
